src/docfx/lib/schema/JsonSchemaValidator.cs

```diff
@@ -253,6 +254,32 @@ private void ValidateDeprecated(JsonSchema schema,  JToken token, List<Error> er
             }
         }
 
+        private void ValidateDependentEnums(JsonSchema schema, JObject map, List<Error> errors)
+        {
+            foreach (var fieldName in schema.DependentEnums.Keys)
+            {
+                if (map.TryGetValue(fieldName, out var fieldValue) && fieldValue.Type == JTokenType.String)
+                {
+                    foreach (var dependentFieldName in schema.DependentEnums[fieldName].Keys)
+                    {
+                        if (map.TryGetValue(dependentFieldName, out var dependentFieldValue))
+                        {
+                            if (dependentFieldValue.Type == JTokenType.String &&
+                                schema.DependentEnums[fieldName][dependentFieldName].TryGetValue((string)dependentFieldValue, out var allowList) &&
+                                Array.IndexOf(allowList, (string)fieldValue) == -1)
```

Comment:
👍 , `Array.IndexOf` avoids LINQ, also update `ValidateEnum` to use `Array.IndexOf` ?